### DISCUSSION TITLE
How to manually define packages using `package-dir` and `packages`?

### DISCUSSION THREAD

**Discussion by ghost** [pypa/setuptools#4513] [Q&A]
Hi there,

I'm using an external dependency to our project that is NOT a Python package.  There is a certain level of hierarchy that I'd like to install as a package so I've been looking at options for adding a pyproject.toml or setup.py to the repo in order to do this.  I seem to have totally failed to find the correct setuptools options but have it working with poetry and I just want to understand what I'm doing wrong.

The directory structure is something like this:

```
remote-repo-name/
├── examples
└── src
    ├── a
    ├── b
    ├── c
    └── d
```

Only directory `b` has the Python that I want to install as a package and I can do this with Poetry using the following pyproject.toml:

```
[project]
name = "remote-repo-name"
version = "0.1.0"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

[tool.poetry]
name = "remote-repo-name"
version = "0.1.0"
authors = ["author"]
description = "description"
packages = [
  { include = "b", from = "src", to = "remote_repo_name" }
]
```

If I do the same with what I think are the correct setuptools commands I get a virtualenv populated with all the directories that have any Python files in.  Including the `src` directory.

```
$ ls -l venv/lib/python3.10/site-packages
src
a
b
c
d
```

I've tried what feels like every possible combination but something like the following is what I thought should work from the examples and documentation I've read:

```
[tool.setuptools.packages.find]
where = ["src"]
include = ["remote_repo_name.b"]
namespaces = true

[tool.setuptools]
package-dir = { "remote_repo_name" = "src" }
```

What is the setuptools equivalent of that one poetry line?

**Comment by abravalheri** [comment]
Hi @shareefj,

I believe that it would look like:

```toml
...
[tool.setuptools]
packages = [
  "b",
  # Manually list all the packages and subpackages you want included (i.e. folders, not individual .py files)
  # "b.b1",
  # "b.b1.b2",
  # ...
]
package-dir = { "" = "src" }
```

***without the `[tool.setuptools.packages.find]`***
(`find` performs an automated scan which is opposed with the expressed desire of manually defining packages).
